examples/mirage/components/signUp/SignUp.tsx

```diff
@@ -0,0 +1,86 @@
+import { KeyboardEvent, useState } from "react";
+import { useClerk, withClerk } from "@clerk/nextjs";
+import type { SignUpResource } from "@clerk/types";
+import { useRouter } from "next/router";
+import { FormLayout } from "../layout/FormLayout";
+import { SignUpEmailStep } from "./SignUpEmailStep";
+import { SignUpCodeStep } from "./SignUpCodeStep";
+import { SignUpFirstNameStep } from "./SignUpFirstNameStep";
+import { SignUpLastNameStep } from "./SignUpLastNameStep";
+import { SignUpUsernameStep } from "./SignUpUsernameStep";
+import { SignUpPhotoStep } from "./SignUpPhotoStep";
+
+enum FormSteps {
+  EMAIL,
+  CODE,
+  FIRST_NAME,
+  LAST_NAME,
+  USERNAME,
+  PHOTO,
+}
+
+function SignUp() {
+  const router = useRouter();
+  const clerk = useClerk();
+
+  const [formStep, setFormStep] = useState(FormSteps.EMAIL);
+  const [createdSessionId, setCreatedSessionId] = useState<string | null>(null);
+
+  const gotoNextStep = () => setFormStep((formStep) => formStep + 1);
+
+  const completeSignUp = (signUp: SignUpResource) => {
+    setCreatedSessionId(signUp.createdSessionId);
+    setFormStep((formStep) => formStep + 1);
+  };
+
+  const completeSignUpAfterUploadingThePhoto = async (photo?: File) => {
+    // Solution 1: Doesn't work as clerk.session is null until the first setSession is called.
+    // Also, Clerk.#updateClient doesn't set the session after authentication via client piggybacking.
+    // if (clerk.session) {
```

Comment:
Note that this approach is discouraged as we don't want users to take session from clerk.session. They should all use `const session = useSession();` that triggers the proper state updates and rerenders.